kibana/templates/deployment.yaml

```diff
@@ -76,11 +85,36 @@ spec:
       imagePullSecrets:
 {{ toYaml .Values.imagePullSecrets | indent 8 }}
     {{- end }}
+      initContainers:
+      - name: configure-kibana-token
+        image: "{{ .Values.image }}:{{ .Values.imageTag }}"
+        imagePullPolicy: "{{ .Values.imagePullPolicy }}"
+        # TODO add retries
+        command:
+          - sh
+          - -c
+          - curl --output {{ template "kibana.home_dir" . }}/config/tokens/{{ template "kibana.fullname" . }}.json --fail -XPOST --cacert {{ template "kibana.home_dir" . }}/config/certs/{{ .Values.elasticsearchCertificateAuthoritiesFile }} -u "$(ELASTICSEARCH_USERNAME):$(ELASTICSEARCH_PASSWORD)" "{{ .Values.elasticsearchHosts }}/_security/service/elastic/kibana/credential/token/{{ template "kibana.fullname" . }}?pretty"
```

Comment:
__question:__ Sorry if it's a dumb question, but I'm not Helm chart/k8s expert: what would happen if the user installs Kibana helm chart, then uninstalls it and try to install it again? Won't this request fail because the init container will try to create token with the same name (`kibana.fullname`)?

Reply:
Now that's a really relevant question. When you uninstall the chart, there is a K8S job that is triggered using Helm `post-delete` hooks that will call Elasticsearch again to remove the token. This way, if we re-install the chart, it can recreate a new token with the same name.

https://github.com/elastic/helm-charts/blob/80e4adab3338fa354626da1dd5965f216ad7d0c1/kibana/templates/job.yaml#L1-L7

https://github.com/elastic/helm-charts/blob/80e4adab3338fa354626da1dd5965f216ad7d0c1/kibana/templates/job.yaml#L23-L31

Reply:
Awesome, thank you for clarifying that!